app/controllers/user/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,28 @@
+class User::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def facebook
+    if user_signed_in?
+      link_current_account_to_facebook
+    else
+      sign_in_with_facebook
+    end
+  end
+
+  private
+
+  def link_current_account_to_facebook
+    # TODO: Implement
+  end
+
+  def sign_in_with_facebook
+    auth = request.env['omniauth.auth']
+    @user = User.from_omniauth(auth)
+    if @user.persisted?
+      sign_in_and_redirect(@user, event: :authentication)
+      set_flash_message(:notice, :success,
+                        kind: t('user.omniauth_callbacks.facebook.kind')) if is_navigational_format?
```

Comment:
Initially I was using it, but turns out that I18n-tasks have complains: `en.user.omniauth_callbakcs.sign_in_with_facebook.facebook.kind` missing, note that it treats `sign_in_with_facebook` as an action.

I think it's a issue of them.